app/views/likes/create.js.erb

```diff
@@ -1,5 +1,7 @@
-$('#like').bind('ajax:success',function() {
+$('.btn').bind('ajax:success',function() {
   $(this).closest('div').find('span').html(<%= "#{@post.likes.count}" %>);
```

Comment:
You html structure could evolve. So everytime you change your structure you will have to update this code as well. 
Instead if above give a unique class to span in current container and then change it's html with jquery

Reply:
I have tried this solution but find method does not work with classes.

Reply:
It will. try again. Debug you code.

Reply:
Done